pkg/gateway/client/identity.go

```diff
@@ -277,6 +277,16 @@ func (c *Client) ensureIdentity(ctx context.Context, tx *gorm.DB, id *types.Iden
 		}
 	}
 
+	for i := range id.AuthProviderGroups {
+		if err := tx.Save(&id.AuthProviderGroups[i]).Error; err != nil {
+			return nil, fmt.Errorf("failed to save group %s: %w", id.AuthProviderGroups[i].ID, err)
+		}
+	}
+
+	if err := tx.Model(id).Association("AuthProviderGroups").Replace(id.AuthProviderGroups); err != nil {
```

Comment:
I think this deserves a comment. Not sure what is going on here.

Reply:
done